### PR TITLE
Speeding up  N3LO matching

### PR DESCRIPTION
**Pull request by giacomomagni**
* [x] Removing explict log(N) form A_Hq parametrisation
* [x] Parametrise A_qqNS removing calls to w=5 harmonics
* [x] Explict shift in the QCD matching order 
* [x] Impose gluon momentum conservation

### CODE REVIEW

File: src/eko/matching_conditions/as3/aHq.py [hunk 1]
```diff
@@ -6,29 +6,35 @@
...
-    Computes the |N3LO| singlet |OME| :math:`A_{Hq}^{S,(3)}(N)`.
-    The experssion is presented in :cite:`Ablinger_2015` (eq 5.1)
+    r"""Computes the |N3LO| singlet |OME| :math:`A_{Hq}^{S,(3)}(N)`.
+    The expression is presented in :cite:`Ablinger_2015` (eq 5.1)
     and :cite:`Blumlein:2017wxd` (eq 3.1).
 
     When using the code, please cite the complete list of references
     available in :mod:`eko.matching_conditions.as3`.
 
+    Note the part proportional to nf^0 includes non trivial weight = 5
```
Comment (felixhekhorn) [inline src/eko/matching_conditions/as3/aHq.py:16]:
```suggestion
    The part proportional to :math:`n_f^0` includes non trivial weight-5
```

File: src/eko/matching_conditions/as3/aHgstfac.py [hunk 2]
```diff
@@ -5,24 +5,24 @@
 
 @nb.njit(cache=True)
 def A_Hgstfac(n, sx, nf):
-    r"""
-    Computes the approximate incomplete part of :math:`A_{Hg}^{S,(3)}(N)`
+    r"""Computes the approximate incomplete part of :math:`A_{Hg}^{S,(3)}(N)`
     proportional to various color factors.
-    The experssion is presented in cite:`ablinger2017heavy` (eq 3.1).
+    The expression is presented in cite:`ablinger2017heavy` (eq 3.1).
```
Comment (felixhekhorn) [inline src/eko/matching_conditions/as3/aHgstfac.py:10]:
```suggestion
    The expression is presented in :cite:`ablinger2017heavy` (eq 3.1).
```

File: tests/eko/test_matching_n3lo.py [hunk 3]
```diff
@@ -143,42 +167,34 @@ def test_Blumlein_3():
             )
 
             np.testing.assert_allclose(
-                aS3[2, 1], ref_val_Hq[L][idx], rtol=2e-5, atol=3e-6
+                aS3[2, 1], ref_val_Hq[L][idx], rtol=2e-5, atol=2e-6
```
Comment (felixhekhorn) [inline tests/eko/test_matching_n3lo.py:170]: this is the only *decrease* in accuracy?

**Review by felixhekhorn — APPROVED** [review]
Seems alright ... as far as I understand it's mostly mathematics changing; did I see correctly that you always improved accuracy, but a single case?

File: tests/eko/test_matching_n3lo.py (hunk 3 above)
Comment (giacomomagni) [reply tests/eko/test_matching_n3lo.py:170]: well Hq was correct also before with the weight 5 harmonics. With the new approximated version we are slightly closer to the known moments, but should be negligible effect.

Comment (felixhekhorn) [reply tests/eko/test_matching_n3lo.py:170]: sorry, successfully confused myself ... of course you're also improving here

**Pull request merged**